### PR TITLE
sioyek: 2.0.0 -> 2.0.0-unstable-2024-09-29

### PR DESCRIPTION
**Pull request by xyven1**
Gives access to many new configuration options, bug fixes, accessibility improvements, and features.
Changelog: https://github.com/ahrm/sioyek/compare/main...development

## Description of changes
- Updated package `version`, `rev`, and `sha256`.
- Added `qtspeech` dependency
- Removed patch as https://github.com/ahrm/sioyek/issues/804 was resolved
- `libsForQt5.callPackage` -> `qt6.callPackage`
- Added xyven1 to package maintainers.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by stephen-huan — APPROVED** [review]
Opened a few PDFs and seems to work.

Result of `nixpkgs-review pr 283813` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sioyek</li>
  </ul>
</details>

**Comment by pca006132** [comment]
We can add `qtwayland` as well, see https://github.com/ahrm/sioyek/issues/867

**Comment by xyven1** [comment]
@pca006132 Do you have a good idea of how to do that? I am happy to add it, but I think first priority is to get this update merged, as Sioyek is more than a year of features behind in nixpkgs.

**Comment by xyven1** [comment]
@podocarp Bumping this PR.

**Comment by pca006132** [comment]
I think you can just add `qtwayland` to `buildInputs` and it will work.

**Comment by pca006132** [comment]
But it seems that visual mark does not work well under Wayland for some reason. Perhaps we should not add `qtwayland` for now.

**Comment by xyven1** [comment]
Been daily driving this for a while, no problems. Still haven't tested on MacOS. I'm on wayland for what its worth.

File: pkgs/applications/misc/sioyek/default.nix
Comment (pluiedev) [inline pkgs/applications/misc/sioyek/default.nix:67]:
```suggestion
      cp pdf_viewer/{prefs,prefs_user,keys,key_user}.config tutorial.pdf sioyek.app/Contents/MacOS/
```

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/sioyek/default.nix
Comment (stephen-huan) [inline pkgs/applications/misc/sioyek/default.nix:26]: Is there a reason this specific commit is used rather than a more recent one?

**Review by stephen-huan — APPROVED** [review]
Still works for me.

Result of `nixpkgs-review pr 283813` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sioyek</li>
  </ul>
</details>

File: pkgs/applications/misc/sioyek/default.nix
Comment (xyven1) [reply pkgs/applications/misc/sioyek/default.nix:26]: because this PR is like 5 months old and if i bump the version then i would need to test it again, and potentially address new issues. This version is still a whole lot better than what is currently in unstable.

Comment (stephen-huan) [reply pkgs/applications/misc/sioyek/default.nix:26]: oh ok. Just fyi I've been using this [overlay](https://github.com/stephen-huan/nixos-config/blob/master/overlays/sioyek.nix) for the latest commit on the development branch and it seems to work

```nix
final: prev:

let
  self = prev.sioyek;
  sioyek' = self.overrideAttrs (previousAttrs: {
    src = final.fetchFromGitHub {
      owner = "ahrm";
      repo = "sioyek";
      rev = "b312fe0b7ae4b8125febef4467480367e0030665";
      sha256 = "sha256-0UCFp52mSEHFuH50FD+CrvmiR9SWfgqm8gUypX7gLG4=";
    };
    buildInputs = previousAttrs.buildInputs ++ [ final.qt6.qtspeech ];
    patches = [ ];
  });
in
{
  sioyek = final.qt6.callPackage sioyek'.override { };
}
```
(though have not tested extensively). I guess one of us can open a new pr once this one lands :p.

Comment (xyven1) [reply pkgs/applications/misc/sioyek/default.nix:26]: Exactly my thought. I just really think this PR should land, its been too damn long. Don't want to mess it up.

**Comment by kilesduli** [comment]
Hi, I noticed that the current version of sioyek in nixpkgs has an issue due to the newer version of mupdf. When reading larger PDFs and using t to jump between sections, there's a memory leak and a white screen. The latest commits in both the development and main branches don’t have this problem, so I’m hoping the PR can be merged to fix this.

**Comment by stephen-huan** [comment]
Perhaps one of us can post on the NixOS discourse in one of the PRs ready for review threads?

I agree that it's been too long for straightforward update.

**Comment by xyven1** [comment]
I will fix the merge conflicts today, so that this is PR is still viable. Again, I *could* update to an even more recent commit, but I (personally) noticed some regressions with the latest version, and feel that the version in this PR has most of the relevant changes (I'm open to any comments, and happy to bump the version if there is something important that got added or fixed).

In regards to adding to review threads, I have added this PR to the review threads in the discord multiple times, but never NixOS discourse, so that is another good option, if someone wants to do that.

**Comment by stephen-huan** [comment]
> I will fix the merge conflicts today, so that this is PR is still viable. Again, I _could_ update to an even more recent commit, but I (personally) noticed some regressions with the latest version, and feel that the version in this PR has most of the relevant changes (I'm open to any comments, and happy to bump the version if there is something important that got added or fixed).

Just out of curiosity, what have you noticed? I'm on [ae733e9](https://github.com/ahrm/sioyek/commit/ae733e927a4bddc49abe81b96a1d96361736ea52) which (was) latest on development branch, and before that, [a3aeca4](https://github.com/ahrm/sioyek/commit/a3aeca4864976a0d09572c58f61da0b2b9896481). To be clear, I'm not suggesting to bump the version, as I also haven't really noticed any improvements.

> In regards to adding to review threads, I have added this PR to the review threads in the discord multiple times, but never NixOS discourse, so that is another good option, if someone wants to do that.

I can do this after the merge conflict is resolved.

**Comment by xyven1** [comment]
> Just out of curiosity, what have you noticed? 

I'm not sure which version I was on (it was the `developement` branch version sometime this past month), but there was a noticeable performance regression on scrolling (the document would not be loaded after scrolling, and would take time to load after being brought into view, which significantly disrupted the reading experience for me). My guess is that this was an intentional memory saving or efficiency change which was either being tested, or needed some polish. YMMV

> I can do this after the merge conflict is resolved.

Appreciated.

**Comment by stephen-huan** [comment]
> The latest commits in both the development and main branches don’t have this problem, so I’m hoping the PR can be merged to fix this.

@kilesduli I just noticed you mentioned the latest commits in development and main fix your issue; have you tested that this PR actually fixes your issue? It's pinned to [36cff8d](https://github.com/ahrm/sioyek/commit/36cff8dbec4bebdab441c21d4128e01fb89b4237) on development which is relatively old.

**Comment by kilesduli** [comment]
> > The latest commits in both the development and main branches don’t have this problem, so I’m hoping the PR can be merged to fix this.
> 
> @kilesduli I just noticed you mentioned the latest commits in development and main fix your issue; have you tested that this PR actually fixes your issue? It's pinned to [36cff8d](https://github.com/ahrm/sioyek/commit/36cff8dbec4bebdab441c21d4128e01fb89b4237) on development which is relatively old.

@stephen-huan  I just tested [36cff8](https://github.com/ahrm/sioyek/commit/36cff8dbec4bebdab441c21d4128e01fb89b4237), and it resolved my issue. However, I found new problem. When scrolling through the table of contents by press <down>, the scrolling is not smooth and frames will drop. Additionally, after clicking on a section to jump to it, the screen freezes, and I have to scroll with the mouse wheel to unfreeze.

For me I would prefer to update.

**Comment by stephen-huan** [comment]
Oh and @xyven1 apologies for the spam, but would it be possible to move this package to the `by-name` hierarchy to take advantage of the new [mergebot](https://github.com/NixOS/nixpkgs-merge-bot)? According to the [documentation](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-merge-pull-requests) and [readme](https://github.com/NixOS/nixpkgs-merge-bot?tab=readme-ov-file#constraints) it seems (1) it needs to be in `pkgs/by-name` (2) the person invoking the bot needs to be a maintainer and (3) the PR is generated by r-ryantm ([nixpkgs-update](https://github.com/nix-community/nixpkgs-update)). I'm not 100% sure whether r-ryantm will correctly identify the version for an unstable package though. 

Not blocking for this PR and I can do it myself. And feel free to add me as a maintainer (`lib.maintainers.stephen-huan`).

**Comment by stephen-huan** [comment]
> @stephen-huan I just tested [36cff8](https://github.com/ahrm/sioyek/commit/36cff8dbec4bebdab441c21d4128e01fb89b4237), and it resolved my issue. However, I found new problem. When scrolling through the table of contents by press , the scrolling is not smooth and frames will drop. Additionally, after clicking on a section to jump to it, the screen freezes, and I have to scroll with the mouse wheel to unfreeze.
> 
> For me I would prefer to update.

> I'm not sure which version I was on (it was the `developement` branch version sometime this past month), but there was a noticeable performance regression on scrolling (the document would not be loaded after scrolling, and would take time to load after being brought into view, which significantly disrupted the reading experience for me).

Hmm, there seems to be a tension here. @xyven1 does your regression still happen on latest development ([965499f](https://github.com/ahrm/sioyek/commit/965499f0acbb1faf4b443b6bca30e7078f944b52))? If so you two can try to find something in the middle, I haven't noticed any problems personally.

**Comment by kilesduli** [comment]
> > @stephen-huan I just tested [36cff8](https://github.com/ahrm/sioyek/commit/36cff8dbec4bebdab441c21d4128e01fb89b4237), and it resolved my issue. However, I found new problem. When scrolling through the table of contents by press , the scrolling is not smooth and frames will drop. Additionally, after clicking on a section to jump to it, the screen freezes, and I have to scroll with the mouse wheel to unfreeze.
> > For me I would prefer to update.
> 
> > I'm not sure which version I was on (it was the `developement` branch version sometime this past month), but there was a noticeable performance regression on scrolling (the document would not be loaded after scrolling, and would take time to load after being brought into view, which significantly disrupted the reading experience for me).
> 
> Hmm, there seems to be a tension here. @xyven1 does your regression still happen on latest development ([965499f](https://github.com/ahrm/sioyek/commit/965499f0acbb1faf4b443b6bca30e7078f944b52))? If so you two can try to find something in the middle, I haven't noticed any problems personally.

I am also on commit [ae733e9](https://github.com/ahrm/sioyek/commit/ae733e927a4bddc49abe81b96a1d96361736ea52), which is the latest commit from three days ago. The problem I just discovered has not been encountered in this commit.

**Comment by stephen-huan** [comment]
> Can i move it to `pkgs/by-name` since it is uses `qt6.callPackage`?

I think so, you take `qt6` as an argument and refer to the qt libraries by `qt6.qtbase`, for instance. See [this package](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/wh/where-is-my-sddm-theme/package.nix).

> Seems as though the latest commit in the dev branch works ok for me (still seems a little worse? but very hard to quantify). I will bump it to that version

Thanks for your sacrifice : )

**Comment by xyven1** [comment]
Got it, I'll give that a shot

**Comment by xyven1** [comment]
Seems to work. Moved to `by-name`

**Comment by xyven1** [comment]
@kilesduli @stephen-huan Can you please run this PR, and see if everything is working as you would expect? If so, I will attempt to invoke merge-bot/

File: pkgs/by-name/si/sioyek/package.nix
Comment (stephen-huan) [inline pkgs/by-name/si/sioyek/package.nix:82]: Feel free to add me as maintainer (`lib.maintainers.stephen-huan`).

**Review by stephen-huan — APPROVED** [review]
Built and opened a few PDFs, looks good to me.

**Comment by kilesduli** [comment]
> @kilesduli @stephen-huan Can you please run this PR, and see if everything is working as you would expect? If so, I will attempt to invoke merge-bot/请你运行这个 PR，看看一切是否按预期工作？如果是的话，我将尝试调用 merge-bot

Looks good to me. Works fine.

**Comment by stephen-huan** [comment]
@xyven1 it seems you might need to add `passthru.updateScript` and `nix-update-script` as an input? See, e.g.


https://github.com/NixOS/nixpkgs/blob/bc947f541ae55e999ffdb4013441347d83b00feb/pkgs/by-name/me/merge-ut-dictionaries/package.nix#L73-L78

and the [contributing README](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#automatic-package-updates).

```nix
passthru.updateScript = nix-update-script {
  extraArgs = [
    "--version"
    "branch"
  ];
};
```

**Comment by xyven1** [comment]
Will do.

**Review by stephen-huan — APPROVED** [review]
Posted to [PRs already reviewed](https://discourse.nixos.org/t/prs-already-reviewed/2617/2016), fingers crossed. Seemed to have a higher merged percentage than [PRs ready for review](https://discourse.nixos.org/t/prs-ready-for-review/3032).

**Comment by xyven1** [comment]
@stephen-huan I very much appreciate the research you did on this to get the PR all shaped up for another merge attempt. When I created the PR I really thought it was just a basic version bump, but I guess because it is to a dev branch it is a little more suspect... Anyways hopefully this finally gets merged.

**Review by phanirithvij — APPROVED** [review]
tested locally `nix run github:nixos/nixpkgs/refs/pull/283813/merge#sioyek`

**Review by kilesduli — APPROVED** [review]

File: pkgs/by-name/si/sioyek/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/si/sioyek/package.nix:49]: We should use --replace-fail now

Comment (xyven1) [reply pkgs/by-name/si/sioyek/package.nix]: Updated.

**Pull request merged**
